### DISCUSSION TITLE
No command line output in RustDesk 1.2.3 (Windows)

### DISCUSSION THREAD

**Discussion by silmaril42** [rustdesk/rustdesk#6377] [General]
### Bug Description

According to [the documentation](https://rustdesk.com/docs/en/client/#command-line-parameters) it should be possible to call `rustdesk.exe` with command line arguments to get certain informations.

[The code](https://github.com/rustdesk/rustdesk/blob/bdc5cded221af9697eb29aa30babce75e987fcc9/src/core_main.rs#L269) also strongly suggests, this should be the case.

I was not able to get any output on `stdout` or `stderr` when trying these commands:
`rustdesk.exe --version`
`rustdesk.exe --get-id`

The command shell returns immediately after those commands. The only output is a new line.

Running the exe with an unknown parameter (eg. `rustdesk.exe --abcdefg`) behaves as if no parameter was provided: the RustDesk UI is opened if it was not open before.

This behaviour is the same, both as a normal or administrative user.

### How to Reproduce

- On Windows, open a command line window (cmd or powershell - doesn't matter which)
- `cd` into the RustDesk installation directory (eg. `"C:\Program Files\RustDesk"`)
- run `rustdesk.exe --version`
- run `rustdesk.exe --get-id`
- Notice that neither command writes any output to the window.

### Expected Behavior

-  `rustdesk.exe --version` should write the current version to `stdout`
-  `rustdesk.exe --get-id` should write the RustDesk ID of the current machine to `stdout`

### Operating system(s) on local side and remote side

Windows 11

### RustDesk Version(s) on local side and remote side

1.2.3

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/31900904/1ffd5252-f56a-4944-85d8-0b51b93935b4)

**Comment by silmaril42** [comment]
I just investigated this further.

Calling `rustdesk.exe` from a python script using `subprocess.run()` returns the expected output strings. 

Using `cygwin`, `Msys2-64` or `Git Bash` (based on `MINGW64`) all work fine and show the expected output.

It seems that only the command line windows inside Windows 11 are not able to transport this information for some reason.

**Reply by TBG-FR** [reply]
Some more information/investigation : 

Once rustdesk is installed, we opened a command line at `C:\Program Files\RustDesk`

```cmd
# This will display/return absolutely nothing
rustdesk --get-id
rustdesk --version

# This will work !
rustdesk --get-id | more
rustdesk --version | more
```

Also, if you do a `rustdesk --password 123456` the password **is changed** but **the value doesn't change in the UI** !

**Comment by danielcshn** [comment]
same problem in Windows 11 (22H2 22621.2506) and version 1.2.1

**Comment by ghost** [comment]
Try this:

```
cd $env:ProgramFiles\RustDesk\
$id = .\rustdesk.exe --get-id | Out-String


```

**Reply by MichaelUray** [reply]
Works for me in Windows PowerShell with Admin privileges:
```
PS C:\Program Files\RustDesk> .\rustdesk.exe --version | Out-String
1.2.3

PS C:\Program Files\RustDesk>
```

**Reply by matthiasg** [reply]
Can you use `--connect` ?

**Comment by matthiasg** [comment]
Piping into another program is quite unintuitive (and in my case not available as I want to trigger it from another program which does not support pipes).

Is there any work being done to fix this ?

**Reply by matthiasg** [reply]
It seems that even with the 'workaround', the program cannot make use of the arguments such as `--connect`. So it seems that its only useful for checking the version.

Sadly https://github.com/rustdesk/rustdesk/issues/8606#issuecomment-2208958059 was closed quickly with a rather snarky comment. 

Can anyone here confirm that using `--connect` is not possible from CLI in windows ?

PS: I am actually confused of the aggressiveness of the comment, wondering what kind of community this is as we are looking into switching from Teamviewer to Rustdesk. Anyone can have a rough day though.

**Reply by Dave-in-TN** [reply]
Don't know if the same will work on Windows, but I was successful on my Mac with the following:

`RustDesk.app --args --connect <device IP>`
